### PR TITLE
Add APNG image support

### PR DESCRIPTION
**Pull request by sagesse-cn**
for #542

### CODE REVIEW

**Comment by sagesse-cn** [comment]
I'm sorry it took so long

**Review by garrettmoon — CHANGES_REQUESTED** [review]
I'm sorry you haven't received a review on this!

Two things to start:
 - Would you mind putting this behind a compiler flag like PIN_WEBP? I'm concerned about adding another check in altertnateRepresentationWithData since it needs to run so frequently. Additionally, I'm not sure how common APNG use is?
- Would you mind adding tests to make sure this is working as expected?

**Comment by sagesse-cn** [comment]
There is only a slight difference between `PINGIFAnimatedImage` and `PINAPNGAnimatedImage`. 
Based a `PINAppleAnimatedImage` would maybe is a better choice.

**Comment by sagesse-cn** [comment]
@garrettmoon This PR has been suspended for a long time, is there anything not modified?

I have updated the PR to the latest code.

**Comment by garrettmoon** [comment]
It has been too long, I'm sorry :(

The CI failures may be related to Xcode updates. I think https://github.com/pinterest/PINRemoteImage/pull/616 may have some fixes there?

File: Source/Classes/AnimatedImages/PINCachedAnimatedImage.m [hunk 1]
```diff
@@ -13,6 +13,7 @@
 #if PIN_WEBP
 #import "PINWebPAnimatedImage.h"
 #endif
+#import "PINAPNGAnimatedImage.h"
```
[the remark this answers is not in the document]
Comment (sagesse-cn) [reply Source/Classes/AnimatedImages/PINCachedAnimatedImage.m:17]: done

File: Source/Classes/AnimatedImages/PINCachedAnimatedImage.m [hunk 2]
```diff
@@ -72,6 +73,9 @@ - (instancetype)initWithAnimatedImageData:(NSData *)animatedImageData
         return [self initWithAnimatedImage:[[PINWebPAnimatedImage alloc] initWithAnimatedImageData:animatedImageData]];
     }
 #endif
+    if ([animatedImageData pin_isAPNG]) {
```
[the remark this answers is not in the document]
Comment (sagesse-cn) [reply Source/Classes/AnimatedImages/PINCachedAnimatedImage.m:79]: done

**Comment by sagesse-cn** [comment]
@garrettmoon it's done

**Comment by garrettmoon** [comment]
Thank you for hanging in there and for the contribution!

**Pull request merged**